data/src/main/java/org/apache/iceberg/data/DataUtil.java

```diff
@@ -0,0 +1,201 @@
+
+    /**
+     * Returns the data files in a partition by listing the partition location.
+     *
+     * For Parquet and ORC partitions, this will read metrics from the file footer. For Avro partitions,
+     * metrics are set to null.
+     *
+     * @param partitionKeys partition key, e.g., "a=1/b=2"
+     * @param uri partition location URI
+     * @param format partition format, avro, parquet or orc
+     * @param spec a partition spec
+     * @param conf a Hadoop conf
+     * @param metricsConfig a metrics conf
+     * @return a List of DataFile
+     */
+  public static List<DataFile> listPartition(Map<String, String> partitionKeys, String uri, String format,
+                                             PartitionSpec spec, Configuration conf, MetricsConfig metricsConfig) {
+    return listPartition(partitionKeys, uri, format, spec, conf, metricsConfig, null);
+  }
+
+    /**
+     * Returns the data files in a partition by listing the partition location.
+     * <p>
+     * For Parquet and ORC partitions, this will read metrics from the file footer. For Avro partitions,
+     * metrics are set to null.
+     * <p>
+     * Note: certain metrics, like NaN counts, that are only supported by iceberg file writers but not file footers,
+     * will not be populated.
+     *
+     * @param partitionKeys partition key, e.g., "a=1/b=2"
+     * @param uri partition location URI
+     * @param format partition format, avro, parquet or orc
+     * @param spec a partition spec
+     * @param conf a Hadoop conf
+     * @param metricsConfig a metrics conf
+     * @param mapping a name mapping
+     * @return a List of DataFile
+     */
+  public static List<DataFile> listPartition(Map<String, String> partitionKeys, String uri, String format,
+                                             PartitionSpec spec, Configuration conf, MetricsConfig metricsConfig,
+                                             NameMapping mapping) {
+    if (format.contains("avro")) {
+      return listAvroPartition(partitionKeys, uri, spec, conf);
+    } else if (format.contains("parquet")) {
+      return listParquetPartition(partitionKeys, uri, spec, conf, metricsConfig, mapping);
+    } else if (format.contains("orc")) {
+      return listOrcPartition(partitionKeys, uri, spec, conf, metricsConfig, mapping);
+    } else {
+      throw new UnsupportedOperationException("Unknown partition format: " + format);
+    }
+  }
+
+  private static List<DataFile> listAvroPartition(Map<String, String> partitionPath, String partitionUri,
+                                                  PartitionSpec spec, Configuration conf) {
+    try {
+      Path partition = new Path(partitionUri);
+      FileSystem fs = partition.getFileSystem(conf);
+      return Arrays.stream(fs.listStatus(partition, HIDDEN_PATH_FILTER))
+              .filter(FileStatus::isFile)
+              .map(stat -> {
+                  // Avro file statistics cannot be calculated without reading the file.
```

Comment:
This does not seem to match the existing logic. If we want to make this change, can we do that in a separate PR?
Also, won't it break the existing filtering logic?

Reply:
@aokolnychyi I changed back to the original value (-1L) and I will open a separate PR to address this issue. 
With -1 set as rowCount will result in [failure|https://github.com/apache/iceberg/blob/a79de571860a290f6e96ac562d616c9c6be2071e/core/src/main/java/org/apache/iceberg/DataFiles.java#L288] when calling DataFiles.Builder.build(). I checked the associated test suite `TestSparkTableUtil`, and it seems that the import of spark tables having data files in avro file format [was not tested|https://github.com/apache/iceberg/blob/a79de571860a290f6e96ac562d616c9c6be2071e/spark2/src/test/java/org/apache/iceberg/spark/source/TestSparkTableUtil.java#L139] at all.